src/Carousel.webmodeler.ts

```diff
@@ -35,26 +35,15 @@ export function getPreviewCss() {
 }
 
 export function getVisibleProperties(props: CarouselContainerProps, visibilityMap: VisibilityMap) {
-    if (props.dataSource === "static") {
-        visibilityMap.imagesEntity = false;
-        visibilityMap.entityConstraint = false;
-        visibilityMap.dataSourceMicroflow = false;
-        visibilityMap.urlAttribute = false;
-    } else if (props.dataSource === "XPath") {
-        visibilityMap.staticImages = false;
-        visibilityMap.dataSourceMicroflow = false;
-    } else if (props.dataSource === "microflow") {
-        visibilityMap.staticImages = false;
-        visibilityMap.imagesEntity = true;
-        visibilityMap.entityConstraint = false;
-    }
+    visibilityMap.dataSourceMicroflow = props.dataSource === "microflow";
+    visibilityMap.entityConstraint = props.dataSource === "XPath";
+    visibilityMap.staticImages = props.dataSource === "static";
+    visibilityMap.imagesEntity = props.dataSource === "microflow";
 
-    if (props.onClickOptions === "doNothing") {
-        visibilityMap.onClickMicroflow = false;
-        visibilityMap.onClickForm = false;
-    } else if (props.onClickOptions === "callMicroflow") {
-        visibilityMap.onClickForm = false;
-    } else if (props.onClickOptions === "showPage") {
-        visibilityMap.onClickMicroflow = false;
-    }
+    visibilityMap.onClickForm = props.onClickOptions === "showPage";
+    visibilityMap.onClickMicroflow = props.onClickOptions === "callMicroflow";
+    visibilityMap.onClickNanoflow = props.onClickOptions === "callNanoflow";
+    visibilityMap.openPageAs = props.onClickOptions === "showPage";
```

Comment:
should be placed after line 43, (as it belongs to the clickform) and the order of the XML